controller/rebuild.go

```diff
@@ -95,24 +96,27 @@ func (c *Controller) VerifyRebuildReplica(address string) error {
 	if err != nil {
 		return err
 	}
-	WOcheckpoint, err := getReplicaCheckpoint(address)
+	WOCheckpoint, err := getReplicaCheckpoint(address)
 	if err != nil {
 		return err
 	}
 	var (
 		indx     int
 		snapshot string
 	)
+	if (WOCheckpoint != "") && (!util.ChainContainsSnapshot(rwChain, WOCheckpoint)) {
+		return fmt.Errorf("WO replica's checkpoint not present in rwReplica chain")
+	}
 	for indx, snapshot = range rwChain {
-		if snapshot == WOcheckpoint {
+		if snapshot == WOCheckpoint {
```

Comment:
does index '0' of rwChain hae snapshot as ""? does index 0 needs special handling?

Reply:
index 0 is head file

Reply:
Index 0 doesn't need to be compared, as head file name might be different across replicas